activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -180,6 +181,10 @@ def supports_common_table_expressions?
         database_version >= "3.8.3"
       end
 
+      def supports_insert_returning?
+        true
```

Comment:
Since the feature detection above and below this are for much older versions of sqlite, I think we need to version detect this one as well:

```suggestion
        database_version >= "3.35.0"
```

Ref https://www.sqlite.org/lang_returning.html

Reply:
Very good point. That was an oversight on my part. Thank you. Fixed and force-pushed